presto-native-execution/presto_cpp/main/PeriodicServiceInventoryManager.cpp

```diff
@@ -85,8 +85,7 @@ void PeriodicServiceInventoryManager::sendRequest() {
       .via(eventBaseThread_.getEventBase())
       .thenValue([this](auto response) {
         auto message = response->headers();
-        if (message->getStatusCode() != http::kHttpAccepted ||
-            message->getStatusCode() != http::kHttpNoContent) {
+        if (message->getStatusCode() != http::kHttpAccepted) {
```

Comment:
Current code allows kHttpNoContent? This change makes kHttpNoContent as a failure case? Is it what we want @amitkdutta? thanks!

Reply:
When the statuscode is 202 (OK), response has no content. Then the logic `message->getStatusCode() != http::kHttpAccepted || message->getStatusCode() != http::kHttpNoContent` fails and announcement is treated as failure.

Check my comment here https://github.com/prestodb/presto/pull/20627#pullrequestreview-1609907289
Basically originally announcement was testing statuscode. I think content checking might be needed for Heartbeatmanagement for multi coordinator, but for general announcement (single coordinator case) its not needed. This PR revert the condition to old state to keep it working.

CC: @pranjalssh

Reply:
"message->getStatusCode() == http::kHttpNoContent" is treated as a failure case now so this is expected? What's the case in code for "When the statuscode is 202 (OK), response has no content."? The comment just makes me a bit confusing. It seems to me that we want to treat no content as a failure case?

Reply:
@xiaoxmeng Yea, so basically only status code OK should be considered as success. Anything else should be considered as failure. In current code, `kHttpNoContent` is considered as success also. We are reverting that change here.